### PR TITLE
Feat/generatepsk

### PR DESCRIPTION
**Pull request by Teufelchen1**
Similar to PR #17 but uses PSK instead of ECC.

### CODE REVIEW

File: Nodes/common.mk [hunk 1]
```diff
@@ -3,4 +3,36 @@ ROOT_DIR = $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
 EXTERNAL_MODULE_DIRS := $(ROOT_DIR)/modules
 RIOTBASE ?= $(ROOT_DIR)/RIOT
 
+JSON_OUTPUT = node.json
+UUID_OUTPUT = uuid.txt
+QRCODE = qrcode.png
+
+# hardcoded wow, many secure, much professional, such cool
+PSK_ID = hardcoded_id
+PSK = hardcoded_psk
```
Comment (leandrolanzieri) [inline Nodes/common.mk:12]: Why not generate them randomly? 
https://linuxhint.com/generate-random-string-bash/

Seeing that link, the kernel provides uuid, maybe useful for the UUID without python ;)

File: Nodes/modules/puzzle_coap/Makefile.dep [hunk 2]
```diff
@@ -4,4 +4,12 @@
 USEMODULE += gcoap
 USEMODULE += cord_ep_standalone
 USEMODULE += uri_parser
+
+USEMODULE += credman
+USEMODULE += credman_load
```
Comment (leandrolanzieri) [inline Nodes/modules/puzzle_coap/Makefile.dep:9]: Don't think you need this one now, as there are no ECC keys to load

File: Nodes/modules/puzzle_coap/puzzle_coap.c [hunk 3]
```diff
@@ -27,24 +30,50 @@ static ssize_t _puzzle_ready_handler(coap_pkt_t *pdu, uint8_t *buf, size_t len,
...
 
+static ssize_t _puzzle_serial_handler(coap_pkt_t *pdu, uint8_t *buf, size_t len, void *ctx);
+
 
 /* CoAP resources. Must be sorted by path (ASCII order). */
 static coap_resource_t _resources[] = {
     { "/node/info", COAP_GET, _puzzle_info_handler, NULL },
     { "/node/maintainance", COAP_PUT, _puzzle_maintainance_handler, NULL },
     { "/node/ready", COAP_PUT, _puzzle_ready_handler, NULL },
+    /*{ "/node/serial", COAP_GET, _puzzle_serial_handler, NULL },*/
```
Comment (leandrolanzieri) [inline Nodes/modules/puzzle_coap/puzzle_coap.c:41]: ?

**Review by leandrolanzieri — CHANGES_REQUESTED** [review]

File: Nodes/modules/puzzle_coap/puzzle_coap.c (hunk 3 above)
Comment (Teufelchen1) [reply Nodes/modules/puzzle_coap/puzzle_coap.c:41]: We haven't debugged yet (or I can't remember) what happens if we expose the same resource twice (over UDP vs DTLS). So I left it as we left it at the Hack'n'ACK.

**Comment by TristanImkn** [comment]
Tested, works:

```
make qr
../common.mk:29: The tool qrencode is not installed. Falling back to an online service...
wget 'https://api.qrserver.com/v1/create-qr-code/?size=300x300&data={"serial":"7c5929aa-00f5-490b-9fe5-ca41c7714986","id":"hardcoded_id","psk": "hardcoded_psk","name":"example_puzzle"}' -O qrcode.png
--2022-06-13 16:06:37--  https://api.qrserver.com/v1/create-qr-code/?size=300x300&data=%7B%22serial%22:%227c5929aa-00f5-490b-9fe5-ca41c7714986%22,%22id%22:%22hardcoded_id%22,%22psk%22:%20%22hardcoded_psk%22,%22name%22:%22example_puzzle%22%7D
Resolving api.qrserver.com (api.qrserver.com)... 88.99.85.235, 195.201.128.178, 159.69.246.187, ...
Connecting to api.qrserver.com (api.qrserver.com)|88.99.85.235|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [image/png]
Saving to: ‘qrcode.png’

qrcode.png                                                                                     [ <=>                                                                                                                                                                                                                                    ]     823  --.-KB/s    in 0s      

2022-06-13 16:06:37 (242 MB/s) - ‘qrcode.png’ saved [823]

```

**Comment by TristanImkn** [comment]
@leandrolanzieri Can you remove your change request ?

**Review by TristanImkn — APPROVED** [review]
Ack

**Review by TristanImkn — APPROVED** [review]

**Pull request merged**